### ISSUE TITLE
Flag alignment on non-retina screens

### ISSUE THREAD

**Issue by jackocnr** [Bluefieldscom/intl-tel-input#178]
If you change the CSS for the .iti-flag background-color to "blue" you will see that we have a problem with a lot of countries (this is seen in Chrome, Safari and FF). I think it's to do with decimal places in pixel values (for width/height/background-position rules).

As a note: this is also a problem on Chrome on retina screens.

UPDATE: I just don't think we should be using floating point pixel values e.g. the US flag has `height: 10.5px` and `background-position: -210px -209.5px` but when it's rendered, there's a gap at the top. It only looks right when you set the values to `height: 10px` and `background-position: -210px -210px`.

UPDATE: I think almost all problems would be solved by taking the floor of the halved value in the retina-size function in sprite-retina-mustache.scss - see http://codepen.io/jackocnr/pen/WbgWbJ?editors=110

**Comment by jackocnr** [comment]
@patw0929 btw I updated that handlebars template in the way I mentioned and it fixed a lot of the problems. But on non-retina screens, you can still see blue in some flags e.g. Andorra, Belgium and Bermuda - all of which are 1px too tall for some reason.

**Comment by jackocnr** [comment]
A ha - I've realised what the problem is - it's back to the build process of building the src/img/flags.png at 2x and then scaling it down - it doesn't preserve the flag borders/padding properly. The proper way to do it is definitely to build everything twice - one for 1x, one for 2x. BUT as we already tried that and thought it didn't look as good, we have to make a decision: do we go with that solution because it's the "right" thing to do - it's much more stable - we know things will always work well in the future, OR do we stick with the current workaround (of adding 2px padding to the retina png, then scaling down, and flooring all pixel values).

UPDATE: actually the issue before was that we were exporting the SVG at the smaller size - what if instead we export the SVG at the 2x size, but then scale down each individual flag (in src/img/flags/) and then build the sprite out of those? Or will we end up with 2 sets of CSS that way? I think that with this solution we should be able to go with zero padding, and the same CSS should just work for both sprites?

**Comment by patw0929** [comment]
Actually I made some try for export different size individually at this branch: https://github.com/patw0929/intl-tel-input/tree/fix-retina-decimal-issue

But sorry for that I have some urgent projects at work recently, and cannot focus on this issue. I think that I'll restart the try on the weekend. Thank you.

**Comment by jackocnr** [comment]
Ok great, thanks Patrick!

**Comment by patw0929** [comment]
On the weekend, I have tried many different ways to generate non-retina & retina flag images & CSS, but it still have some issue here.

Just like [this branch](https://github.com/patw0929/intl-tel-input/tree/fix-retina-decimal-issue)'s solution, I have generated width/height/background-position of flags avoid float number in non-retina version, and finally it looks good and seems solved the alignment problem.

But unfortunately, in retina (@2x) version, some of them still have some alignment issue. I also generate flag sprites (double padding) and set it as half size in SCSS. 

I think maybe the main problem is that - some of the flags have "odd number" in width/height (to keep the aspect ratio), and if it's in half, the float number is shown again...

Still think other way to solve it. I think I'm stuck here. :/  
Do you have other thought of it?

**Comment by jackocnr** [comment]
Hi Patrick, yer I've thought about it and I agree - if the retina images have an odd number in the width/height then there's no way to use the same CSS for both sprites. But this must be a common problem - no retina images should contain odd dimensions. I don't understand why grunt-responsive-images is outputting images with odd dimensions. The only solution is to find a tool that will output the @2x images so they all have even dimensions. There must be a way to do that. What do you think?

**Comment by patw0929** [comment]
Same branch: https://github.com/patw0929/intl-tel-input/tree/fix-retina-decimal-issue

I found a way to solve this issue. [evenizer](https://github.com/katapad/evenizer) can make the images in odd dimensions convert to even dimensions. But the way they use is add a transparent background to the original image, so I made some revise: force resize the image to even dimensions. [my fork.. waiting the pull request to merge](https://github.com/patw0929/evenizer/tree/add-resize-mode)

Add this task to intl-tel-input's Grunt task, and install evenizer. (my fork)
```
npm install -g git://github.com/patw0929/evenizer.git#add-resize-mode
```

Run the build again, then it seems like that we solved this issue.
